File: webapi/Models/HR/CompanyEmployeeTax.cs
using webapi.Models.System;

namespace webapi.Models.HR
{
    public record CompanyEmployeeTax
    {
        public int Id { get; set; }
        public int EmployeeId { get; set; }
        public Employee? Employee { get; set; }
        public int SysPaymentTypeId { get; set; }
        public SysPaymentType? SysPaymentType { get; set; }
        public DateOnly DisbursementAccrualDate { get; set; }
        public DateOnly CreationDate { get; set; }
    }

    public class CompanyEmployeeTaxDTO
    {
        public int EmployeeId { get; set; }
        public int SysPaymentTypeId { get; set; }
        public DateOnly DisbursementAccrualDate { get; set; }
    }
}

File: reactapp/src/pages/EndMonth/models/CompanyEmployeeTax.ts


export interface CompanyEmployeeTax {
    id: int
    employeeId: int,
    sysPaymentTypeId: int,
    disbursementAccrualDate: string,
    creationDate: string
}


export const defaultCompanyEmployeeTaxUpdate: CompanyEmployeeTax = {
    id: 0,
    employeeId: 0,
    sysPaymentTypeId: 0,
    disbursementAccrualDate: "",
    creationDate: ""
}



export interface CompanyEmployeeTaxInsert {
    employeeId: int,
    sysPaymentTypeId: int,
    disbursementAccrualDate: string,
}


export const createDefaultCompanyEmployeeTax = (employeeId: int): CompanyEmployeeTaxInsert => {
    return {
        employeeId: employeeId,
        sysPaymentTypeId: 0,
        disbursementAccrualDate: "",
    }
}